### PR TITLE
Disable stderr output in Release builds

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - Port PR https://github.com/openvinotoolkit/openvino/pull/10048

### CODE REVIEW

**Review by rkazants — COMMENTED** [review]
I didn't find any stderr in TF FE. Why do we need this change in TF FE?

File: src/frontends/tensorflow/src/tensorflow.cpp [hunk 1]
```diff
@@ -2,6 +2,8 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
+#include <google/protobuf/stubs/logging.h>
+
```
Comment (rkazants) [inline src/frontends/tensorflow/src/tensorflow.cpp:6]: please do not include protobuf dependencies here. Can we face segfault issue on OVTF side, if they use different protobuf version.

**Review by rkazants — CHANGES_REQUESTED** [review]
Looks risky. Explanation is below in the comment.

**Comment by rkazants** [comment]
@ilya-lavrenov, @mvafin, could you please take a look? Can we face the same issue with protobuf segfault after this change, if OVTF uses different protobuf version?

**Comment by ilyachur** [comment]
> @ilya-lavrenov, @mvafin, could you please take a look? Can we face the same issue with protobuf segfault after this change, if OVTF uses different protobuf version?

@rkazants Let's differentiate  public and private API. Here we use protobuf API inside our component which uses static protobuf, so it means that we change behaviour only our static protobuf which we are using inside. From my point of view it shouldn't affect OVTF or other external components.

File: src/frontends/tensorflow/src/tensorflow.cpp (hunk 1 above)
Comment (ilyachur) [reply src/frontends/tensorflow/src/tensorflow.cpp:6]: Why? it is our internal code. We use static protobuf inside our components

**Comment by rkazants** [comment]
> > @ilya-lavrenov, @mvafin, could you please take a look? Can we face the same issue with protobuf segfault after this change, if OVTF uses different protobuf version?
> 
> @rkazants Let's differentiate public and private API. Here we use protobuf API inside our component which uses static protobuf, so it means that we change behaviour only our static protobuf which we are using inside. From my point of view it shouldn't affect OVTF or other external components.

Ilya, please let me explain. We had the same situation previously when we statically linked protobuf to TF FE and it leads to the segfault on the OVTF side. The problem is that the same process uses two different versions of the protobuf that can have different environment initializations that can conflict and lead to the segfault.
Currently all protobuf stuff is not directly linked to TF FE. Instead we use a wrapper for it `DecoderProto` that includes protobuf stuff. It saved for OVTF scenario that they have own `DecoderProto` and do not create `DecoderProto` object used in MO+TF FE case.

Regards,
Roman

File: src/frontends/tensorflow/src/tensorflow.cpp (hunk 1 above)
Comment (rkazants) [reply src/frontends/tensorflow/src/tensorflow.cpp:6]: the problem is explained below

**Comment by ilyachur** [comment]
> > > @ilya-lavrenov, @mvafin, could you please take a look? Can we face the same issue with protobuf segfault after this change, if OVTF uses different protobuf version?
> > 
> > 
> > @rkazants Let's differentiate public and private API. Here we use protobuf API inside our component which uses static protobuf, so it means that we change behaviour only our static protobuf which we are using inside. From my point of view it shouldn't affect OVTF or other external components.
> 
> Ilya, please let me explain. We had the same situation previously when we statically linked protobuf to TF FE and it leads to the segfault on the OVTF side. The problem is that the same process uses two different versions of the protobuf that can have different environment initializations that can conflict and lead to the segfault. Currently all protobuf stuff is not directly linked to TF FE. Instead we use a wrapper for it `DecoderProto` that includes protobuf stuff. It saved for OVTF scenario that they have own `DecoderProto` and do not create `DecoderProto` object used in MO+TF FE case.
> 
> Regards, Roman

Nice to know it. So yes it means that the change for OV TF is incorrect. So I will revert it.

**Review by rkazants — APPROVED** [review]

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**